### PR TITLE
Bunch of tracking pipeline changes

### PR DESCRIPTION
**Pull request by DaWelter**
Hi @sthalik , @GO63-samara 

I would like to discuss a few changes. Commit by commit:

* Combine Accela with Hamilton filter: We can compute the delta as angle-axis pair like in your Hamilton filter @GO63-samara , and apply the gain function from Accela. The extra wide "zoom deadzone" could also be integrated easily. Didn't do that yet.
Hence I propose to combine both filters and take the best of both worlds. On that occassion I would also delete the Kalman filter (which I wrote) because it just isn't as good.

* Fix compile error: Not very interesting for you, I guess. It's just that the ORT API didn't like the constness of the allocator which it had because it was used from a const-function.

* Fix 180 deg roll: I noticed the neuralnet tracker outputs roll offset by +/- 180 deg. This shall be the fix.

* apply_zero_post first ...: My tentative attempt to fix the yaw-pitch coupling. So I moved this `apply_zero_post` forward in the processing chain and changed it to work like the VR 360 centering option. The idea being that we do a proper 3d coordinate change. So the head pose afterwards is measured wrt this new centered frame including the subsequently computed euler angles.
Should help with cameras mounted at an angle. Intended use is simple, the user attains a centered pose, notes the raw tracker output and fills these values into the zero-pose fields. As a side effect, opentrack starts roughly centered, due to the zero-pose persistence.
I tried to keep the function in the old place, but I didn't get the right results. I suppose it's because the mapping curves are applied beforehand. I don't understand it though tbh. Probably my change breaks a bunch of other options. Hence it's not meant as the ultimate solution. And I also realize that it's very redundant with the center function when VR 360 is used. @sthalik , I would like to understand what the other center modes are for. I think maybe we could remove some options except the VR 360 center mode. Any ideas?

* Try new model: A new neural net which, after (brief) testing, I like more than the original. It provides the uncertainty estimates which are needed for the extra filtering in the tracker itself. I observed the uncertainty estimates increase when eyes get closed actually. They drive the size of a deadzone. During testing I got a more stable view which was especially noticable when zooming in, trying to focus on targets. Maybe you can give it a try.

### CODE REVIEW

File: logic/pipeline.cpp [hunk 1]
```diff
@@ -390,9 +390,20 @@ Pose pipeline::maybe_apply_filter(const Pose& value) const
...
-    for (int i = 0; i < 6; i++)
-        value(i) += m(i).opts.zero * (m(i).opts.invert ? -1 : 1);
 
+    const auto q = QQuaternion::fromEulerAngles(value[Pitch], value[Yaw], -value[Roll]);
+    const auto center = QQuaternion::fromEulerAngles(
+        m(Pitch).opts.zero * (m(Pitch).opts.invert ? -1 : 1),
+        m(Yaw).opts.zero * (m(Yaw).opts.invert ? -1 : 1),
+        m(Roll).opts.zero * (m(Roll).opts.invert ? 1 : -1)
+    );
+    const auto v = (center.conjugated() * q).toEulerAngles();
```
Comment (sthalik) [inline logic/pipeline.cpp:400]: It's good in principle, just add a separate setting for it and leave `apply_zero_pos` as it was.

**Review by sthalik — CHANGES_REQUESTED** [review]

File: filter-accela/ftnoir_filter_accela.cpp [hunk 2]
```diff
@@ -127,9 +130,40 @@ void accela::filter(const double* input, double *output)
         });
 #else
     do_deltas(&deltas[Yaw], &output[Yaw], [this](double x) {
-        return spline_rot.get_value_no_save(x);
+        returnspline_rot.get_value_no_save(x);
```
[the remark this answers is not in the document]
Comment (DaWelter) [reply filter-accela/ftnoir_filter_accela.cpp:133]: Oops

File: filter-accela/ftnoir_filter_accela.cpp [hunk 3]
```diff
@@ -148,12 +182,15 @@ void accela::filter(const double* input, double *output)
         return spline_pos.get_value_no_save(x);
     });
 
+    for (unsigned i=0; i<3; ++i)
```
[the remark this answers is not in the document]
Comment (DaWelter) [reply filter-accela/ftnoir_filter_accela.cpp:185]: 3, position only. I had to split it because the new rotation stuff needs other formulas.

**Comment by DaWelter** [comment]
> Zero pos is only meant for fine-tuning the cockpit position
Oh I see. Well, there is enough room in the dialog to put another option ...

Anyhow, your suggestions are more work intensive than I hoped for ;-) Regardless I feel like doing it some time in future.

When you have some time to play, please test out my new neural net.

**Comment by DaWelter** [comment]
Closed because as a showcase of unrelated things this was never intended to be merged.
